### DISCUSSION TITLE
Wrong registry used in pnpm install

### DISCUSSION THREAD

**Discussion by NEWESTERS** [pnpm/pnpm#4972] [Q&A]
Our company uses private Nexus repo as npm registry (e.g. `http://mycompany.com/repository/registry-A`), it is specified in `.npmrc` like this:
```
registry=http://mycompany.com/repository/registry-A
//mycompany.com/repository/:_auth:my-auth-token
```

Some time ago `registry-A` was  removed and now i should use `registry-B`, so `.npmrc` has changed:
```
registry=http://mycompany.com/repository/registry-B
//mycompany.com/repository/:_auth:my-auth-token
```

But when i try to `pnpm install` on my machine, pnpm uses `registry-A` instead of `registry-B` despite the fact that i had removed all mentions of `registry-A` from my machine.

It happens only when pnpm should resolve version for dependency (if it is missing in pnpm-lock.yaml).

When my colleague, that never used `registry-A`, tries to install dependencies, pnpm uses correct registry.

Also when i try to install dependencies in docker environment, all goes well.

Seems like pnpm somehow "remembered" `registry-A` for domain `http://mycompany.com`  and use it only on my machine, but i can not figure out why.

**Comment by craigmiller160** [comment]
This question, for me, is literally that meme on XKCD where someone finds a StackOverflow post with their exact problem, but nobody answered it and OP never responded. I'm having literally the exact same problem...

**Comment by craigmiller160** [comment]
Ugh, I found the answer and I'm going to put it here. For global packages, it has its own hidden settings. On my mac, I found the file at /Users/craigmiller/Library/pnpm/global/5/node_modules/.modules.yaml. You need to modify this to point at the correct new registry.

**Reply by mbtools** [reply]
FYI, corresponding files on Windows installations with corepack:

`C:\Users\<username>\AppData\Local\node\corepack\pnpm\8.14.0\dist\node_modules\.modules.yaml`
`C:\Users\<username>\AppData\Local\node\corepack\v1\pnpm\8.14.0\dist\node_modules\.modules.yaml`

But those are pointing to the correct registry 🤷 . I will keep looking...

**Comment by NEWESTERS** [comment]
2 years ago the only solution that i have is monkey patching pnpm.

Hardcoding correct registry url inside `fetchFromRegistry` method does the trick.  🧙
